src/timebar/timebar.js

```diff
@@ -29,7 +28,15 @@ const ONE_DAY_MS = 1000 * 60 * 60 * 24
 class Timebar extends Component {
   constructor() {
     super()
+    const toggleImmediate = (immediate) => {
+      this.setState((state) => ({
+        immediate,
+      }))
+    }
+    this.interval = null
     this.state = {
+      immediate: false,
+      toggleImmediate,
```

Comment:
do we need to have the toggleImmediate inside the state? I think is not a good pattern and it isn't necessary

Reply:
I agree and I fixed it. Although I'm wondering: why do you think React docs encourage doing that? : https://reactjs.org/docs/context.html#updating-context-from-a-nested-component (see app.js)

Reply:
I guess because they pass the entire state to the context... thanks for the change!